lib/report.js

```diff
@@ -150,6 +195,113 @@ class Report {
     return mergeProcessCovs(v8ProcessCovs)
   }
 
+  /**
+   * If --all is supplied we need to fetch a list of files that respects
+   * include/exclude that will be used to see the coverage report with
+   * empty results for unloaded files
+   * @returns {Promise.<Array.<string>>}
+   */
+  async getFileListForAll () {
+    return this.exclude.globSync(this.src).reduce((allFileList, file) => {
+      const srcPath = join(this.src, file)
+      allFileList.set(srcPath, false)
+      return allFileList
+    }, new Map())
+  }
+
+  /**
+   * Iterates over the entries of `allFilesMap` and where an entries' boolean
+   * value is false, generate an empty coverage record for the file in question.
+   * @param {Map<string, boolean>} allFilesMap where the key is the path to a file
+   * read by `--all` and the boolean value indicates whether a coverage record
+   * for this file was found.
+   * @param {CoverageMap} coverageMap A coverage map produced from v8's output.
+   * If we encounter an unloaded file, it is merged into this CoverageMap
+   * @returns {Promise.<undefined>}
+   * @private
+   */
+  async _createEmptyRecordsForUnloadedFiles (allFilesMap, coverageMap) {
+    for (const [path, seen] of allFilesMap.entries()) {
+      // if value is false, that means we didn't receive a coverage
+      // record. Create and merge an empty record for the file
+      if (seen === false) {
+        const emptyCoverageMap = await this._getEmpyCoverageResultForFile(path)
+        coverageMap.merge(emptyCoverageMap)
+      }
+    }
+  }
+
+  /**
+   * Uses `v8toIstanbul` to create CoverageMapData for a file with all statements,
+   * functions and branches set to unreached
+   * @param {string} fullPath
+   * @returns {Promise.<CoverageMapData>}
+   * @private
+   */
+  async _getEmpyCoverageResultForFile (fullPath) {
+    const converter = v8toIstanbul(fullPath, this.wrapperLength)
+    await converter.load()
+    const initialCoverage = converter.toIstanbul()
+    this._setCoverageMapToUncovered(Object.values(initialCoverage)[0])
```

Comment:
I would be tempted to pull this into its own file, e.g., `src/get-empty-coverage.js`.